docs/architecture/adr-035-rosetta-api-support.md

```diff
@@ -0,0 +1,225 @@
+# ADR 035: Rosetta API Support
+
+## Context
+
+[Rosetta API](https://www.rosetta-api.org/) is an open-source specification and set of tools developed by Coinbase to standardise blockchain interactions.
+
+Through the use of a standard API for integrating blockchain applications it will
+
+* Be easier for a user to interact with a given blockchain
+* Allow exchanges to integrate new blockchains quickly and easily
+* Enable application developers to build cross-blockchain applications such as block explorers, wallets and dApps at considerably lower cost and effort.
+
+## Decision
+
+It is clear that adding Rosetta API support to the Cosmos SDK will bring value to all the developers and Cosmos SDK based chains in the ecosystem. How it is implemented is key.
+
+The driving principles of the proposed design are:
+
+1. **Extensibility:** it must be as riskless and painless as possible for application developers to set-up network configurations to expose Rosetta API-compliant services.
+2. **Long term support:** This proposal aims to provide support for all the supported Cosmos SDK release series.
+3. **Cost-efficiency:** Backporting features from `master` to the various stable branches of Cosmos SDK is a cost that needs to be reduced.
+
+This ADR will describe how this will be achieved.
+
+1. There will be an external repo called [cosmos-rosetta-gateway](https://github.com/tendermint/cosmos-rosetta-gateway) for the implementation of the core Rosetta API features, particularly:
+   a. The types and interfaces. This separates design from implementation detail.
+   b. Some core implementations: specifically, the `Service` functionality as this is independent of the Cosmos SDK version.
+2. Due to differences between the Cosmos release series, each series will have its own specific API implementations of `Network` struct and `Adapter` interface.
+3. There will be two options for starting an API service in applications:
+   a. API shares the application process
+   b. API-specific process.
+
+This document below will describe each design component of the proposed architecture.
+
+## Architecture
+
+### 1. The External Repo
+
+As section will describe the proposed external library, including the service implementation, plus the defined types and interfaces.
+
+#### Service
+
+`Service` is a simple `struct` that is started and listens to the port specified in the options. This is meant to be used across all the Cosmos SDK versions that are actively supported.
+
+The constructor follows:
+
+`func New(options Options, network Network) (*Service, error)`
+
+#### Types
+
+`Service` accepts an `Options` `struct` that holds service configuration values, such as the port the service would be listening to:
+
+```golang
+type Options struct {
+	Port uint32
```

Comment:
Why not just provide a listen address? This is typically what you do. In addition, these options can be extended to provide CORS, HTTP w/r timeouts, etc...

Reply:
Right! Good point.